src/core/src/op/util/scatter_elements_update_base.cpp

```diff
@@ -88,46 +88,29 @@ bool op::util::ScatterElementsUpdateBase::is_supported_index_input_element_type(
     }
 }
 
-bool op::util::ScatterElementsUpdateBase::evaluate_lower(ov::TensorVector& output_values) const {
+bool util::ScatterElementsUpdateBase::evaluate_lower(ov::TensorVector& output_values) const {
     OV_OP_SCOPE(util_ScatterNDUpdate_evaluate_lower);
     return get_input_tensor(1).has_and_set_bound() && ov::default_lower_bound_evaluator(this, output_values);
 }
 
-bool op::util::ScatterElementsUpdateBase::evaluate_upper(ov::TensorVector& output_values) const {
+bool util::ScatterElementsUpdateBase::evaluate_upper(ov::TensorVector& output_values) const {
     OV_OP_SCOPE(util_ScatterNDUpdate_evaluate_upper);
     return get_input_tensor(1).has_and_set_bound() && ov::default_upper_bound_evaluator(this, output_values);
 }
 
-bool op::util::ScatterElementsUpdateBase::evaluate_label(TensorLabelVector& output_labels) const {
+bool util::ScatterElementsUpdateBase::evaluate_label(TensorLabelVector& output_labels) const {
     OV_OP_SCOPE(util_ScatterNDUpdate_evaluate_label);
 
-    OPENVINO_SUPPRESS_DEPRECATED_START
     return ov::default_label_evaluator(this, {0, 2}, output_labels);
-    OPENVINO_SUPPRESS_DEPRECATED_END
 }
 
-OPENVINO_SUPPRESS_DEPRECATED_START
-int64_t op::util::ScatterElementsUpdateBase::get_normalized_axis(const HostTensorVector& inputs) const {
-    OPENVINO_ASSERT(inputs[3]->get_element_type().is_integral_number(), "axis element type is not integral data type");
+int64_t util::ScatterElementsUpdateBase::get_normalized_axis(const TensorVector& inputs) const {
+    const auto& axis_input = inputs[3];
+    OPENVINO_ASSERT(axis_input.get_element_type().is_integral_number(), "axis element type is not integral data type");
 
-    OPENVINO_SUPPRESS_DEPRECATED_START
-    int64_t axis = host_tensor_2_vector<int64_t>(inputs[3])[0];
-    OPENVINO_SUPPRESS_DEPRECATED_END
-    const auto& input_rank = get_input_partial_shape(0).rank();
-    int64_t normalized_axis = axis;
-
-    if (normalized_axis < 0) {
-        if (input_rank.is_static()) {
-            OPENVINO_SUPPRESS_DEPRECATED_START
-            normalized_axis = ov::normalize_axis(this, axis, input_rank);
-            OPENVINO_SUPPRESS_DEPRECATED_END
-        } else {
-            OPENVINO_SUPPRESS_DEPRECATED_START
-            normalized_axis = ov::normalize_axis(this, axis, static_cast<int64_t>(inputs[0]->get_shape().size()));
-            OPENVINO_SUPPRESS_DEPRECATED_END
-        }
-    }
-    return normalized_axis;
+    const auto axis = v0::Constant{axis_input}.cast_vector<int64_t>()[0];
+    const auto data_rank = static_cast<int64_t>(inputs[0].get_shape().size());
```

Comment:
This is not exactly same as rank is always taken from tensor shape not from node (can be different).
Unless this check is done before call this function.

Reply:
Right. Previous "stronger" normalization restored.